### PR TITLE
Implemented nanmean

### PR DESCRIPTION
**Pull request by MohamedElgharbawy**
Also implemented `__invert__` (`~` operator).

### CODE REVIEW

**Review by elibol — COMMENTED** [review]
Looks good! Left a few minor comments.

File: nums/core/array/application.py [hunk 1]
```diff
@@ -1087,3 +1087,20 @@ def _vec_from_oids(self, oids, shape, block_shape, dtype):
 
     def random_state(self, seed=None):
         return NumsRandomState(self.system, seed)
+
+    def nanmean(self, a: BlockArray, axis=None, keepdims=False, dtype=None):
+        if a.dtype not in (float, np.float32, np.float64):
```
[the remark this answers is not in the document]
Comment (MohamedElgharbawy) [reply nums/core/array/application.py:1092]: Yes.

File: nums/core/array/application.py [hunk 2]
```diff
@@ -1087,3 +1087,20 @@ def _vec_from_oids(self, oids, shape, block_shape, dtype):
...
+            a = a.astype(np.float64)
+
+        num_summed = self.sum(~a.ufunc("isnan"), axis=axis, dtype=a.dtype, keepdims=keepdims)
+
+        if num_summed.ndim > 0:
```
[the remark this answers is not in the document]
Comment (MohamedElgharbawy) [reply nums/core/array/application.py:1097]: I'll add an edge case to return `nan` when `num_summed == 0`.

File: tests/numpy/test_nan_reduction.py [hunk 3]
```diff
@@ -38,9 +38,9 @@ def test_nan_reductions(nps_app_inst):
     for block_shape in block_shapes:
         ba = ba.reshape(block_shape=block_shape)
         np_arr = ba.get()
-        op_params = ["nanmax", "nanmin", "nansum"]
+        op_params = ["nanmax", "nanmin", "nansum", "nanmean"]
         axis_params = [None, 0, 1]
-        # keepdims=True is a bug with some nan functions
+        # keepdims=True is buggy with some nan functions
```
[the remark this answers is not in the document]
Comment (MohamedElgharbawy) [reply tests/numpy/test_nan_reduction.py:43]: I'm a bit fuzzy on it, but when we were trying some nan stat numpy functions it wasn't working. It's probably due to us just calling reduce though, instead of doing a proper implementation. I'll remove this comment.

File: nums/core/settings.py [hunk 4]
```diff
@@ -67,5 +67,6 @@
     "amax",
     "nanmax",
     "nanmin",
-    "nansum"
+    "nansum",
+    "nanmean"
```
[the remark this answers is not in the document]
Comment (MohamedElgharbawy) [reply nums/core/settings.py:71]: Nope, removed.
